server/service/device_client.go

```diff
@@ -0,0 +1,56 @@
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+type DeviceClient struct {
+	*baseClient
+}
+
+func (dc *DeviceClient) request(verb string, path string, query string, responseDest interface{}) error {
+	var bodyBytes []byte
+	request, err := http.NewRequest(
+		verb,
+		dc.url(path, query).String(),
+		bytes.NewBuffer(bodyBytes),
+	)
+
+	if err != nil {
+		return err
+	}
+
+	response, err := dc.http.Do(request)
+	if err != nil {
+		return fmt.Errorf("%s %s: %w", verb, path, err)
+	}
+	defer response.Body.Close()
+
+	return dc.parseResponse(verb, path, response, responseDest)
+}
+
+// NewDeviceClient instantiates a new client to perform requests against device endpoints
+func NewDeviceClient(addr string, insecureSkipVerify bool, rootCA string) (*DeviceClient, error) {
```

Comment:
I believe it makes sense for `NewDeviceClient` to receive the `token` and set it in the field. It's not supposed to change throughout the life of the `DeviceClient`.